players/g4_player.py

```diff
@@ -114,26 +228,45 @@ def _get_target_cells(self, size: int, cog: cell, xmax: int) -> list[cell]:
         # from(yunlan): I think I've confused x and y coords, the target shape
         # actually has buckets facing downwards... To fix later
 
-        buckets = math.floor((size - 2) / 4)
+        buckets = math.floor((size - 3) / 7)
         arm_cells_cnt = 1 + buckets
         wall_cells_cnt = size - arm_cells_cnt
+        inner_wall_cells_cnt = math.ceil(wall_cells_cnt / 2)
+        outer_wall_cells_cnt = math.floor(wall_cells_cnt / 2)
 
         _, y_cog = cog
-        wall_top    = (y_cog + math.floor((wall_cells_cnt - 1) / 2))
-        wall_bottom = (y_cog - math.ceil((wall_cells_cnt - 1) / 2))
-        wall_cell_ys = np.linspace(wall_bottom, wall_top, wall_cells_cnt, True, dtype=int)
+        inner_wall_top    = (y_cog + math.floor((inner_wall_cells_cnt - 1) / 2))
+        inner_wall_bottom = (y_cog - math.ceil((inner_wall_cells_cnt - 1) / 2))
+        inner_wall_cell_ys = np.linspace(inner_wall_bottom, inner_wall_top, inner_wall_cells_cnt, True, dtype=int)
+
+        outer_wall_top    = (y_cog + math.floor((outer_wall_cells_cnt - 1) / 2))
+        outer_wall_bottom = (y_cog - math.ceil((outer_wall_cells_cnt - 1) / 2))
+        outer_wall_cell_ys = np.linspace(outer_wall_bottom, outer_wall_top, outer_wall_cells_cnt, True, dtype=int)
 
         arm_top    = (y_cog + 3 * math.floor((arm_cells_cnt - 1) / 2))
         arm_bottom = (y_cog - 3 * math.ceil((arm_cells_cnt - 1) / 2))
         arm_cell_ys = np.linspace(arm_bottom, arm_top, arm_cells_cnt, True, dtype=int)
 
+        # TODO: statically using xmax=51 can form a haircomb when A=3
+        # actually using xmax, self._reshape returns moves that cause
+        # separation, need to debug why, might be worth writing helpers
+        # to plot the target_cells, retract_cells, and to_occupy_cells
+        #
+        # Note that by using xmax as the x-value of wall cells, and
+        # xmax + 1 as the x-value of arm_cells will automatically cause
+        # the ameoba to shift right, but unsure why it's not working right
+        # now :(
         wall_cells = [
+            ( xmax % 100, y % 100 )
+            for y in inner_wall_cell_ys
+        ] + [
             ( (xmax - 1) % 100, y % 100 )
-            for y in wall_cell_ys
+            for y in outer_wall_cell_ys
         ]
 
+
         arm_cells = [
-            ( xmax % 100, y % 100 )
+            ( (xmax + 1) % 100, y % 100 )
             for y in arm_cell_ys
         ]
```

Comment:
bucket arm may never be formed if we are always moving (an artifact of using `xmax+1` as the x-value of the bucket arm in the target shape)

<img width="1233" alt="image" src="https://user-images.githubusercontent.com/25857014/203234634-a8fbc4bc-0280-470e-a4e3-01aadc1ccf8e.png">

Reply:
(d90e644) disable moving for now, forming the comb shape seems to work for different sizes - haven't tested extensively. Below is one for A=15, and density=0.

<img width="971" alt="image" src="https://user-images.githubusercontent.com/25857014/203235166-d1d5b402-3ec7-4e2b-82da-00457e88773a.png">